src/prisma/schema.prisma

```diff
@@ -1625,3 +1640,91 @@ model BlockCount {
 
   @@index([spaceId])
 }
+
+model UserNotificationRecord {
+  id                    String                 @id @default(uuid()) @db.Uuid
+  createdAt             DateTime               @default(now())
+  createdBy             String                 @db.Uuid
+  deletedAt             DateTime?
+  seenAt                DateTime?
+  spaceId               String                 @db.Uuid
+  space                 Space                  @relation(fields: [spaceId], references: [id], onDelete: Cascade)
+  userId                String                 @db.Uuid
+  user                  User                   @relation(fields: [userId], references: [id], onDelete: Cascade)
+  bountyNotifications   BountyNotification[]
+  pageNotifications     PageNotification[]
+  cardNotifications     CardNotification[]
+  postNotifications     PostNotification[]
+  proposalNotifications ProposalNotification[]
+  voteNotifications     VoteNotification[]
+}
```

Comment:
this is the only part that's weird to me about this design: each user notification has a many to one relationship, but in practice it should only have one linked bounty. i think we can just map this into something more useful when retrieving a user's notifications

Reply:
I think this should be fine. These cross relations don't show up on the base record anyways in Typescript unless we explicitly add them since they're joins.